subgraphs/uniswap-v3/src/common/price/price.ts

```diff
@@ -139,46 +144,105 @@ export function findUSDPricePerToken(
+          false
+        );
+        // get the derived whitelist token in pool
+        const whitelistTokenValueLocked = poolAmounts.inputTokenBalances[
+          whitelistTokenIndex
+        ].times(whitelistToken.lastPriceUSD!);
+
+        // Check if it meets criteria to update prices.
+        if (
+          whitelistTokenValueLocked.gt(largestWhitelistTokenValue) &&
+          whitelistTokenValueLocked.gt(
+            NetworkConfigs.getMinimumLiquidityThreshold()
+          )
+        ) {
+          // Calculate new price of a token and TVL of token in this pool.
+          const newPriceSoFar = poolAmounts.tokenPrices[
+            whitelistTokenIndex
+          ].times(whitelistToken.lastPriceUSD as BigDecimal);
+
+          const newTokenTotalValueLocked = convertTokenToDecimal(
+            token._totalSupply,
+            token.decimals
+          ).times(newPriceSoFar);
+
+          // If price is too high, skip this pool
+          if (newTokenTotalValueLocked.gt(BIGDECIMAL_TEN_BILLION)) {
+            log.warning("Price too high for token: {} from pool: {}", [
+              token.id.toHexString(),
+              pool.id.toHexString(),
+            ]);
+            continue;
           }
+
+          // Set new price and largest pool for pricing.
+          largestWhitelistTokenValue = whitelistTokenValueLocked;
+          priceSoFar = newPriceSoFar;
         }
       }
     }
   }
+
+  // Buffer token pricings that would cause large spikes on the protocol level
+  const protocol = getOrCreateProtocol();
+  const tokenTVLDelta = absBigDecimal(
+    priceSoFar
+      .times(convertTokenToDecimal(token._totalSupply, token.decimals))
+      .minus(token._totalValueLockedUSD)
+  );
+  const protocolTVLPercentageDelta = absBigDecimal(
+    safeDiv(tokenTVLDelta, protocol.totalValueLockedUSD)
+  );
+  if (protocolTVLPercentageDelta.gt(BIGDECIMAL_FIVE_PERCENT)) {
+    log.warning("Price too high for token: {} from pool: {}", [
+      token.id.toHexString(),
+      priceSoFar.toString(),
+    ]);
+    token._largeTVLImpactBuffer += 1;
+    return token.lastPriceUSD!;
+  }
+
+  // Uncomment this after the current production version is fully synced and re-deploy to make the oracle more robust
+  if (!token.lastPriceUSD || token.lastPriceUSD!.equals(BIGDECIMAL_ZERO)) {
+    return priceSoFar;
+  }
+
+  // If priceSoFar 10x greater or less than token.lastPriceUSD, use token.lastPriceUSD
+  // Increment buffer so that it allows large price jumps if seen repeatedly
+  if (
+    priceSoFar.gt(token.lastPriceUSD!.times(BIGDECIMAL_TWO)) ||
+    priceSoFar.lt(token.lastPriceUSD!.div(BIGDECIMAL_TWO))
+  ) {
+    if (token._largePriceChangeBuffer < PRICE_CHANGE_BUFFER_LIMIT) {
+      token._largePriceChangeBuffer = token._largePriceChangeBuffer + 1;
+      return token.lastPriceUSD!;
+    }
+  }
+
+  token._largePriceChangeBuffer = 0;
+  token._largeTVLImpactBuffer = 0;
+
+  token.save();
   return priceSoFar; // nothing was found return 0
```

Comment:
couldn't this be >0 now?

Reply:
Yeah - that was a dumb comment, it should always be > 0